### DISCUSSION TITLE
TextureRegion extra zoom levels and auto update.

### DISCUSSION THREAD

**Discussion by Kiaazad** [godotengine/godot-proposals#4800] [Editor]
Working with my texture atlas and 9 patch frames, I've bumped into two small issues that can be solved easily.
1- TextureRegion window can use a couple of extra zoom levels, to ease working on pixel critical images. Even on my 24" monitor, pixel peeping and adjusting the lines can be difficult.
2- When adjusting my `patch margins` the lines in TextureRegion don't update, I have to click another node and click back to see the changes.

**Comment by Calinou** [comment]
Adding more zoom levels sounds good to me. Feel free to open a pull request for this :slightly_smiling_face: 

> 2- When adjusting my patch margins the lines in TextureRegion don't update, I have to click another node and click back to see the changes.

This sounds like a bug. Please open an issue on the main Godot repository.

**Edit:** Issue opened: https://github.com/godotengine/godot/issues/62694

**Reply by Kiaazad** [reply]
Do I add a pull request and just suggest the extra zoom levels? I always assumed that pull requests must include the code to do or fix something.
I'll report the bug soon, just have to download the latest version and make sure it's not fixed yet.

**Reply by Calinou** [reply]
> Do I add a pull request and just suggest the extra zoom levels? I always assumed that pull requests must include the code to do or fix something.

Indeed, pull requests are what you make when you already have a working implementation. My message isn't specifically addressed to you, but to anyone reading this discussion :slightly_smiling_face:

**Comment by Kiaazad** [comment]
I've checked godot 4 and tried splicing my atlas, it has gotten worse. Instead of streamlining the workflow, even more clicks is needed to do the same thing now.
- The region editor now opens on top and blocks all clicks, including the region number fields.
- Still have to double click the resource to edit it, this already leads to tons of mistakes when editing multiple cuts of the atlas, one click should do the job. I know this comes in handy when dragging things into the inspector, but that feature is not used as often, and can be done with a simple pin button for inspector instead of forcing the player to double click constantly.
- Having to open and close the region editor instead of staying open at the bottom adds at least two additional clicks that are unnecessary 
- When duplicating a resource, the one that I have duplicated remains selected, instead of switching to the newly created one, this often leads to editing the wrong resource.
- The zoom is still 1 level short for precise work on a 24" monitor. Lacks several level for the smaller display of a surface tablet and touch controls.

Many of the mentioned issues exist in godot 3.5.1 as well.
Godot should get a couple of UX designers on board and try to improve the interface's usability. As a rule of thumb, less clicks is better than more.
